docs/template_plugin/tests/functional/op_reference/comparison.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+
+#include "base_reference_test.hpp"
+#include "ngraph_functions/builders.hpp"
+
+namespace ComparisonOpsRefTestDefinitions {
+struct RefComparisonParams {
+    template <class IT, class OT>
+    RefComparisonParams(const ngraph::helpers::ComparisonTypes comp, const ngraph::PartialShape& input_shape1, const ngraph::PartialShape& input_shape2,
+                        const ngraph::element::Type& iType, const ngraph::element::Type& oType, const std::vector<IT>& iValues1,
+                        const std::vector<IT>& iValues2, const std::vector<OT>& oValues)
+        : comparisonType(comp),
+          pshape1(input_shape1),
+          pshape2(input_shape2),
+          inType(iType),
+          outType(oType),
+          inputData1(CreateBlob(iType, iValues1)),
+          inputData2(CreateBlob(iType, iValues2)),
+          refData(CreateBlob(oType, oValues)) {}
+    ngraph::helpers::ComparisonTypes comparisonType;
+    ngraph::PartialShape pshape1;
+    ngraph::PartialShape pshape2;
+    ngraph::element::Type inType;
+    ngraph::element::Type outType;
+    InferenceEngine::Blob::Ptr inputData1;
+    InferenceEngine::Blob::Ptr inputData2;
+    InferenceEngine::Blob::Ptr refData;
+};
+
+class ReferenceComparisonLayerTest : public testing::TestWithParam<RefComparisonParams>, public CommonReferenceTest {
```

Comment:
Please rebase your branch to the latest master, because we merged PR with small refactoring.